pkg/firestore/publisher.go

```diff
@@ -80,9 +83,15 @@ type subscriptionsCacheEntry struct {
 func NewPublisher(config PublisherConfig, logger watermill.LoggerAdapter) (*Publisher, error) {
 	config.setDefaults()
 
-	client, err := firestore.NewClient(context.Background(), config.ProjectID, config.GoogleClientOpts...)
-	if err != nil {
-		return nil, err
+	var client *firestore.Client
+	if config.CustomFirestoreClient != nil {
+		client = config.CustomFirestoreClient
+	} else {
+		var err error
+		client, err = firestore.NewClient(context.Background(), config.ProjectID, config.GoogleClientOpts...)
```

Comment:
btw maybe we can pass also context now?

Reply:
it's probably a good idea how it's used by client constructor 🤔 for google cloud pub/sub we are also not providing that, so maybe whatever